src/python/pants/core/goals/lint.py

```diff
@@ -447,6 +445,7 @@ def batch_by_size(
             (batch, partition.metadata)
             for partitions in partitions_list
             for partition in partitions
+            if partition.elements
```

Comment:
eh? was this hit in the real world?

Reply:
I was seeing empty tuples flow get passed to `partition_sequentially` in tests. I think they're harmless as long as the fallback case in the `key` lambda does `str(x)`, but it also seemed like unintentional behavior so I added this filter. Don't feel strongly about it, can delete it if you want.

Reply:
Yeah let's chuck it. Our other batching code doesn't have this, so unless we're sure it's needed let's not confuse ourselves.